File: front-end/src/domain/entity/truck.ts
import { UllUUID } from '../../utils/ull-uuid';
import { Utils } from '../../utils/utils';
import { Plate } from '../valueobject/plate';
import { Brand } from '../valueobject/brand';
import { Driver } from '../valueobject/driver';

export class Truck {
    
    private _id: UllUUID = UllUUID.random();
    private _plate: Plate;
    private _brand: Brand;
    private _driver: Driver;
    private _isUnload: boolean;

    constructor(id: UllUUID = UllUUID.random(), plate: Plate, brand: Brand, driver: Driver, isUnload: boolean) {
        this._id = id;
        this._plate = plate;
        this._brand = brand;
        this._driver = driver;
        this._isUnload = isUnload;
    }
    
    get id(): UllUUID {
        return this._id;
    }
    
    set id(id: UllUUID) {
        this._id = id;
    }
    
    set plate(plate: Plate) {
        this._plate = plate;
    }

    get plate(): Plate {
        return this._plate;
    }

    get brand(): Brand {
        return this._brand;
    }

    set brand(brand: Brand) {
        this._brand = brand;
    }

    get driver(): Driver {
        return this._driver;
    }

    set driver(driver: Driver) {
        this._driver = driver;
    }

    get isUnload(): boolean {
        return this._isUnload;
    }    

    set isUnload(isUnload: boolean) {
        this._isUnload = isUnload;
    }

    public static random(): Truck {
        return new Truck(
            UllUUID.random(),
            Plate.random(),
            Brand.random(),
            Driver.random(),
            Utils.randomInt(0, 1) === 0 ? false : true
        );
    }

    public toJson(): any {
        return {
            "idTruck": this.id.toString(),
            "plate": this.plate.plate,
            "brand": this.brand.brand,
            "driver": this.driver.driver,
            "unload": this.isUnload
        };
    }

    public static fromJson(json: any): Truck {
        return new Truck(
            new UllUUID(json.idTruck),
            new Plate(json.plate),
            new Brand(json.brand),
            new Driver(json.driver),
            json.unload
        );
    }
}

File: producer-accesses/src/main/java/es/ull/domain/entity/Truck.java
package es.ull.domain.entity;

import java.util.UUID;
import es.ull.domain.valueobject.Plate;
import es.ull.domain.valueobject.Brand;
import es.ull.domain.valueobject.Driver;

import es.ull.utils.TruckUtils;

public class Truck {

    private UUID id;
    private Plate plate;
    private Brand brand;
    private Driver driver;
    private boolean isUnload;

    public UUID getId() {
        return this.id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public Plate getPlate() {
        return this.plate;
    }

    public void setPlate(Plate plate) {
        this.plate = plate;
    }

    public Brand getBrand() {
        return this.brand;
    }

    public void setBrand(Brand brand) {
        this.brand = brand;
    }

    public boolean isUnload() {
        return this.isUnload;
    }

    public void setUnload(boolean isUnload) {
        this.isUnload = isUnload;
    }

    public Driver getDriver() {
        return this.driver;
    }

    public void setDriver(Driver driver) {
        this.driver = driver;
    }

    public static Truck random() {
        Truck truck = new Truck();
        truck.setId(UUID.randomUUID());
        truck.setPlate(Plate.random());
        truck.setBrand(Brand.random());
        truck.setUnload(TruckUtils.generateRandomNumber(1) % 2 == 0);
        truck.setDriver(Driver.random());
        return truck;
    }

    public String toJson() {
        return "{" +
            "\"idTruck\":\"" + this.getId().toString() + "\", " +
            "\"plate\":\"" + this.getPlate().getPlate() + "\", " +
            "\"brand\":\"" + this.getBrand().getBrand() + "\", " +
            "\"driver\":\"" + this.getDriver().getDriver() + "\", " +
            "\"unload\":" + this.isUnload() + "}";
    }
}
